vnext/ReactUWP/Modules/DeviceInfoModule.cpp

```diff
@@ -20,25 +20,35 @@ DeviceInfo::DeviceInfo(const std::shared_ptr<IReactInstance> &reactInstance) : m
 
 void DeviceInfo::update() {
   auto displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+  m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
   winrt::Windows::UI::ViewManagement::UISettings uiSettings;
 
   auto const &window = winrt::Windows::UI::Xaml::Window::Current().CoreWindow();
-
+  m_width = window.Bounds().Width;
+  m_height = window.Bounds().Height;
   m_dimensions = folly::dynamic::object(
       "windowPhysicalPixels",
-      folly::dynamic::object("width", window.Bounds().Width)("height", window.Bounds().Height)(
-          "scale", static_cast<int>(displayInfo.ResolutionScale()) / 100)("fontScale", uiSettings.TextScaleFactor())(
-          "densityDpi", displayInfo.LogicalDpi()))(
+      folly::dynamic::object("width", m_width * m_scale)("height", m_height * m_scale)(
+          "scale", m_scale)("fontScale", uiSettings.TextScaleFactor())("densityDpi", displayInfo.LogicalDpi()))(
       "screenPhysicalPixels",
       folly::dynamic::object("width", displayInfo.ScreenWidthInRawPixels())(
-          "height", displayInfo.ScreenHeightInRawPixels())(
-          "scale", static_cast<int>(displayInfo.ResolutionScale()) / 100)("fontScale", uiSettings.TextScaleFactor())(
+          "height", displayInfo.ScreenHeightInRawPixels())("scale", m_scale)("fontScale", uiSettings.TextScaleFactor())(
           "densityDpi", displayInfo.LogicalDpi()));
+
+    m_dpiChangedRevoker = displayInfo.DpiChanged(winrt::auto_revoke, [this](const auto &, const auto &) {
+        auto displayInfo = winrt::Windows::Graphics::Display::DisplayInformation::GetForCurrentView();
+        m_scale = static_cast<float>(displayInfo.ResolutionScale()) / 100;
+        m_dimensions["windowPhysicalPixels"]["scale"] = m_scale;
+        m_dimensions["windowPhysicalPixels"]["densityDpi"] = displayInfo.LogicalDpi();
+        m_dimensions["screenPhysicalPixels"]["scale"] = m_scale;
+        m_dimensions["screenPhysicalPixels"]["densityDpi"] = displayInfo.LogicalDpi();
+        updateRootElementSize(m_width, m_height);
+  });
 }
 
 void DeviceInfo::updateRootElementSize(float width, float height) {
```

Comment:
>float width, float height [](start = 39, length = 25)

These values aren't making their way into m_width/m_height, which seems like a bug.  First let's discuss whether or not we use the window bounds.

Reply:
Thanks for the catch, made some additional changes on my high DPI machine for testing but forgot to copy it over.

---
In reply to: [371594664](https://github.com/microsoft/react-native-windows/pull/3956#discussion_r371594664) [](ancestors = 371594664)